--- Helper.php.orig	2023-06-12 11:18:14.571543276 +0000
+++ Helper.php	2023-06-12 11:19:14.287601284 +0000
@@ -57,7 +57,12 @@
 		$internalPath = $mount->getInternalPath($absoluteDir);
 
 		$originalLocationsCache = null;
-		$dirContent = $storage->getCache()->getFolderContents($mount->getInternalPath($view->getAbsolutePath($dir)));
+
+		// SURF patch - Trashbin Helper for functional accounts
+		//$dirContent = $storage->getCache()->getFolderContents($mount->getInternalPath($view->getAbsolutePath($dir)));
+		$user = \OCA\group_lookup\AppInfo\getFunctionalAccount($user);
+		$dirContent = $storage->getCache()->getFolderContentsById(\OCA\group_lookup\AppInfo\getFileTrashbinFileId($absoluteDir, $user));
+
 		foreach ($dirContent as $entry) {
 			// construct base fileinfo entries
 			$entryName = $entry->getName();
